### ISSUE TITLE
INFO: TraceTogether to be open sourced

### ISSUE THREAD

**Issue by Carglglz** [ehidra/contact-tracer#1]
Hi, this may be of help if the source code is released 👍 
[TraceTogether to be open sourced](https://mothership.sg/2020/03/tracetogether-installed-open-source/)

[BlueTrace-Manifesto](https://tracetogether.zendesk.com/hc/en-sg/articles/360044883814-BlueTrace-Manifesto)

**Comment by Rnorback** [comment]
This is great @Carglglz. Thanks for sharing.

**Comment by paulmaunders** [comment]
Thank you! We are closely following this and will consider how we can help!

**Comment by waqarrashid33** [comment]
Looking forward to it.

**Comment by smorrel1** [comment]
The Singapore Government Technology Agency replied and will connect us with their project team. That team is busy so not sure when they will get back. Will keep you posted.

**Comment by baradhili** [comment]
@smorrel1 yup.. at the same time several other governments are busy developing their own...
I was on an inter-agency conference call between GOARN, WHO, CDC, EAN, and a whole grab bag of more acronyms.. They were receptive of our efforts and will provide some advice when we are at a stage we can demo things to them...

**Comment by baradhili** [comment]
I'm seeing quite a few western govts spin up their teams to develop clones of this - so our "market" might be the US (who are barely contact tracing at all) and low income locations... My contact in the response team for Jamaica is quite interested

**Comment by anupdey99** [comment]
I am looking forward to it.

**Comment by DavidStodolsky** [comment]
TraceTogether doesn't provide the instant notification that is necessary according to the Oxford report. The system proposed by the Oxford team requires a trusted central authority which will limit adoption, because trust in authorities is at an all time low. The model I proposed solves these problems:

http://cosmism.blogspot.com/2020/03/from-tracing-to-preemption.html

**Comment by waqarrashid33** [comment]
Isn't there some concesus based blockchain technology that can help us
remove the reliance on one central authority.

**Comment by DavidStodolsky** [comment]
Blockchain tech will probably be needed for a comprehensive solution, but there are simpler ways to do it right off with decent security:

The security architecture outlined in the published paper would make it impossible to report users. The transmitted info doesn’t have any user ID info at all. These are just random numbers labelling risky meetings. 

The only time that user info might be exposed is when a user is tested positive. That user may then allow the random numbers from their phone to be broadcast. When one of those numbers matches a number stored in someone else’s phone, the receiver might try to guess which person they had met recently was the infected person. However, every time that a person thought to have been exposed is required to take a test, we also issue a test request to a random person. This allows us to see whether the targeted tests are superior to random testing of people. So, a request for a test would be just as likely to be a “false alarm” as a true targeted test.  

The bottom line is that if you want to have a fresh health certificate you must get tested, if one fails to arrived on time. That test could be anonymous. So, the security architecture doesn’t require any exposure of identity info whatsoever.

**Comment by baradhili** [comment]
Lets avoid blockchain.. especially since the target "market" will be low income countries who can't replicate Singapore stuff themselves.. it can be far simpler as well.. we link a UUID to a phone number - thats all we keep on the backend (or maybe a country as well).. Thats a really low privacy risk, though protecting that list will be crucial as it will become a target for scammers (just like Singapore is fighting). The user's device carries a list of UUIDs it has encountered on a decaying list - if they test positive they can transmit that list to the tracing team - it might be worth having a look at the WHO's [Go.Data](https://www.who.int/godata) tool which is what they will be managing outbreaks with in many countries. It might be possible to transmit directly to that system.

But keep in mind.. delay = deaths - we need a MVP out ASAP.. lets avoid feature creep and keep to the minimum

**Comment by cmorillo** [comment]
Closing this as is being monitor on our slack channel

**Comment by bengregori** [comment]
Hi @cmorillo can you provide link to the slack channel? I would also like to be included when BlueTrace/TraceTogether is opened!

**Comment by paulmaunders** [comment]
Please follow us on Twitter and we will send you a DM: https://twitter.com/app_tracer

**Comment by DavidStodolsky** [comment]
Video meeting 7pm (CEST), Monday, 6 April, 2020 -

https://groups.io/g/MedicalEthics/message/55

It seems the more a project is likely to contain the pandemic, the less support it is currently getting. This is even more true for low income settings. Safe Paths Privacy Kit has  a #africa_latam_carib_strategy Slack channel, but the psychological assumptions that have been accepted by Safe Paths limit their impact. The assumption is that active alerts would be too stressful for users, so users must "pull" the data to see if they are likely infected.

I am trying to create a team that would look at developing a user-driven contagion management system outside of the current institutional structures. Feel free to join us for a chat that will probably last for an hour. Hopefully, we can do this every week for a while.

**Comment by baradhili** [comment]
Oh I'm linked to the response team for Jamaica as well... 

but keep in mind.. fancy tech is pointless in low income countries if there is low penetration of high tech... remember an entire village might be using a single feature phone as a way to call people.. its not like everyone is carrying a smart phone like in high income countries..

**Comment by baradhili** [comment]
@bengregori BlueTrace/TraceTogether won't be opened to the public.. please forget about that.. its only being opened for governments and under very strict restrictions..

**Comment by DavidStodolsky** [comment]
> remember an entire village might be using a single feature phone as a way to call people

I think that my 1997 model would make it possible to use any phone with text messaging. Tracing one person actively can likely find at least a few others passively - manual contact tracing. Even a small part of the population being traced could reduce R0 enough.

https://drive.google.com/file/d/13YswwQ5KrDITQu4m_XB-RhJYhBvt9gTB/view

https://docs.google.com/document/d/1kZj44-Gvau1zeVedkbSQxRihdRywnDqKAbu_heYEEb8/edit#


Given the cost of this pandemic, supplying cheap phones or donated ones might be seen as a smart political move or smart marketing move by a company like Samsung. Taiwan is now sending millions of masks to the US, etc. Why not millions of phones to Jamaica?

**Comment by DavidStodolsky** [comment]
> its only being opened for governments and under very strict restrictions. 

They may fear embarrassment due to flaws in the code. Also, it is now clear that pandemic control requires instant notification, which they have rejected. 


https://www.covid-watch.org/articles

Singapore just released an app that performs some contact tracing using Bluetooth proximity networks. However, our initial analysis suggests their privacy model is susceptible to attack because they did not take advantage of Bluetooth's potential for decentralization.

https://trello-attachments.s3.amazonaws.com/5e750fa118cc121ce836e525/5e76ae82e927b944ff3ce999/22c00e17fbc36c0dd6961d08f83bbca4/Centralized_and_Decentralized_Contact_Tracing_Security_Models.pdf

**Comment by DavidStodolsky** [comment]
The latest I have heard from Singapore is that containment has failed and they are imposing lockdowns on areas where there is community spread. This is a lot better than closing down the whole country, but it is an indication that TraceTogether is not the answer.

**Comment by DavidStodolsky** [comment]
Breaking news is a total lockdown of Singapore and Australia is likely next.

https://7news.com.au/lifestyle/health-wellbeing/singapore-loses-control-of-coronavirus-outbreak-announces-month-long-lockdown--c-958235

**Comment by baradhili** [comment]
@DavidStodolsky Tracetogether was never "the answer".. it wasn't intended to be... Singapore s in "the dance" where they need to lockdown from time to time to stop outbreaks...

BTW 7 news = fox.. so they specialise is overly dramatic "reporting"..
